src/wwwroot/controllers/MyBillingInformationCtrl.js

```diff
@@ -0,0 +1,193 @@
+          vm.paymentMetodInfo.expDate = month + '/' + year;
+        }
+
+        vm.paymentMetodInfo.cc.brand = data.paymentMethod.creditCard.cardBrand;
+        vm.paymentMetodInfo.cc.number = data.paymentMethod.creditCard.cardNumber.replaceAll('#','*');
+        vm.paymentMetodInfo.secCode.mask = getMaskSecCodByBrand(vm.paymentMetodInfo.cc.brand);
+        vm.paymentMetodInfo.secCode.number = vm.paymentMetodInfo.secCode.mask.replaceAll('9','*');
+      }
+
+      vm.paymentMethodLoader = false;
+    }
+
+    function fillCreditCardProperties(newNumber) {
+      vm.cc.brand = getCreditCardBrand(newNumber);
+      vm.cc.mask = getMaskByBrand(vm.cc.brand);
+      vm.secCode.mask = getMaskSecCodByBrand(vm.cc.brand);
+      vm.cc.number = newNumber;
+    }
+
+    function getMaskSecCodByBrand(cardBrand){
+      return secCodeMasksByBrand[cardBrand];
+    }
+
+    function getMaskByBrand(cardBrand){
+      return creditCardMasksByBrand[cardBrand];
+    }
+
+    function getCreditCardBrand(creditCardNumber) {
+    	// start without knowing the credit card brand
+    	var result = "unknown";
+
+    	// first check for MasterCard
+    	if (/^5[1-5]/.test(creditCardNumber)) {
+        	result = "mastercard";
+    	}
+    	// then check for Visa
+    	else if (/^4/.test(creditCardNumber)) {
+    		result = "visa";
+    	}
+    	// then check for AmEx
+    	else if (/^3[47]/.test(creditCardNumber)) {
+    		result = "amex";
+    	}
+    	return result;
+    }
+
+    function checkExpDate(input) {
+      var currentDate = new Date();
+      var currentYear = currentDate.getFullYear().toString().slice(2);
+      var currentMonth = currentDate.getMonth() + 1;
+      if (input.$modelValue && input.$modelValue.length == 4) {
+        var year = input.$modelValue.slice(2);
+        var month = input.$modelValue.slice(0,2);
+        if (year < currentYear || month > 12) {
+          utils.setServerValidationToField($scope, $scope.form.expDate, 'ilegal_date');
+        }
+        if (year == currentYear && month < currentMonth) {
+          utils.setServerValidationToField($scope, $scope.form.expDate, 'ilegal_date');
+        }
+      }
+    }
+
+    var onExpectedError = function (rejectionData) {
+      return true;
+    };
+
+    function submitPaymentMethod(form) {
+      if (!utils.validateCreditCard(vm.cc.number)) {
+        utils.setServerValidationToField($scope, form.cardNumber, 'invalid_card_number');
+      }
+      if (!form.$valid) {
+        return;
+      }
+      vm.cc.parsedCcNumber = utils.replaceAllCharsExceptLast4(vm.cc.number);
+      vm.secCode.ParsedNumber = utils.replaceAllCharsExceptLast4(vm.secCode.number);
+      vm.viewExpDate = form.expDate.$viewValue;
+
+      var paymentMethod = {
+        creditCard: {
+          cardNumber: vm.cc.number,
+          verificationCode: vm.secCode.number,
+          expiryDate: vm.expDate,
+          cardHoldersName: vm.cardHolder
```

Comment:
I talked with @jhoffman-ms about including client-side validations using the same server-side validations but It is ok not to include that to be consistent with other Relay webapp sections